### ISSUE TITLE
[Enhancement] Possibility for a Lettuce variant of "ebean-redis"?

### ISSUE THREAD

**Issue by Mechite** [ebean-orm/ebean#3225]
Personally I find Lettuce to be the modern client. Jedis brings in dependencies such as `gson`, `org.json:json`, but is also just not a very well made client, whereas both Lettuce and Redisson are arguably much better quality clients.

Proposed naming scheme could be any of the following:
- Renaming `ebean-redis`:
  - `ebean-jedis`
  - `ebean-lettuce`
  - `ebean-redisson`
- Suffix instead:
  - `ebean-redis` - aggregator/bundle
  - `ebean-redis-jedis`
  - `ebean-redis-lettuce`
  - `ebean-redis-redisson`

In any case, it should be trivial to at least support Lettuce so we have one high-quality implementation to target. I'm unsure what the correct abstraction for a connection would be to support both regular, cluster, etc connections, but I'm thinking instead of providing/injecting our own client connection (and having to deal with the complexity of this), there can instead be connection configuration and ebean connect to the server automatically, using cluster if needed depending on config etc.

**Comment by SentryMan** [comment]
I also don't like jedis

**Comment by rob-bygrave** [comment]
> Personally I find Lettuce to be the modern client

Yeah, no problem with that. Are you going to build / contribute a Ebean-lettuce module?  Happy to have a look at that and rename things as appropriate when we have that etc.

**Comment by Mechite** [comment]
> > Personally I find Lettuce to be the modern client
> 
> Yeah, no problem with that. Are you going to build / contribute a ebean-lettuce module? Happy to have a look at that and rename things as appropriate when we have that etc.

I can absolutely do that (when I make the time) - I'll maintain it as a fork of this repository (and ebean-redis) and try to mirror the same functionality, will link here when we have that

**Comment by rbygrave** [comment]
Cool. We can close this for the meantime and re-open when you get to it.

**Comment by noboomu** [comment]
[We built one](https://github.com/s1-solutions/ebean-lettuce), please take a look if it's still of interest.

**Comment by noboomu** [comment]
I would be honored if it passes your muster sir.

**Comment by noboomu** [comment]
I would be honored if it passes your muster sir.
